build/debug_keybase/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/korovkin/limiter"
+	"gopkg.in/yaml.v2"
+
+	"github.com/pokt-network/pocket/app/client/keybase"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+)
+
+const (
+	// NOTE: This is the number of validators in the private-keys.yaml manifest file
+	numValidators = 999
+
+	// Increasing this number is linearly proportional to the amount of RAM required for the debug client to start and import
+	// pre-generated keys into the keybase. Beware that might cause OOM and process can exit with 137 status code.
+	// 4 threads takes 350-400MiB from a few tests which sounds acceptable.
+	debugKeybaseImportConcurrencyLimit = 4
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Println("Usage: go run main.go <source_yaml> <target_folder>")
+		return
+	}
+	sourceYamlPath := os.Args[1]
+	targetFolderPath := os.Args[2]
+
+	privateKeysYamlBytes, err := ioutil.ReadFile(sourceYamlPath)
+	if err != nil {
+		fmt.Printf("Error reading source_yaml: %v\n", err)
+		return
+	}
+	sourceYamlHash := md5.Sum(privateKeysYamlBytes)
+	hashString := fmt.Sprintf("%x.md5", sourceYamlHash)
+	hashFilePath := filepath.Join(targetFolderPath, hashString)
+	targetFilePath := filepath.Join(targetFolderPath, "debug_keybase.bak")
+	_, err = os.Stat(hashFilePath)
```

Comment:
Not completely necissary but we have the function `FileExists` in `shared/utils/file_utils.go`

```go
// Check file at the given path exists
func FileExists(path string) (bool, error) {
	_, err := os.Stat(path)
	if err == nil {
		return true, nil
	}
	if errors.Is(err, fs.ErrNotExist) {
		return false, nil
	}
	return false, err
}
```

If you don't decide to change to use this see the comment below

Reply:
This is great! Thanks for pointing that out:

Much better
![image](https://user-images.githubusercontent.com/29378614/227233773-3fb7199f-d19c-45be-a58e-04ba313069bf.png)

🙏